### PR TITLE
001 view job roles

### PR DESCRIPTION
**Pull request by ashleighB31**

### CODE REVIEW

**Review by ReannaMcE — APPROVED** [review]

File: src/controllers/RoleController.ts [hunk 1]
```diff
@@ -0,0 +1,12 @@
+import express from "express";
+import { getJobRoles } from "../services/JobRoleService"
+
+export const getAllJobRoles = async (req: express.Request, res: express.Response): Promise<void> => {
+    try {
+        res.render('jobRolesList.html', { roles: await getJobRoles() });
+        console.log(await getJobRoles())
```
Comment (shaunganley) [inline src/controllers/RoleController.ts]: can remove log

File: src/services/JobRoleService.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import axios, { AxiosResponse } from "axios";
+import { JobRolesResponse } from "../models/JobRolesResponse";
+
+//axios.defaults.baseURL = process.env.API_URL || 'http://localhost:8080/';
+
+//export const URL: string = "/api/job-roles/";
+
+export const getJobRoles = async (): Promise<JobRolesResponse[]> => {
+    try {
+        const response: AxiosResponse = await axios.get("http://localhost:8080/api/job-roles");
```
Comment (shaunganley) [inline src/services/JobRoleService.ts]: uncomment these lines and use them in the axios call

File: src/models/JobRoles.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+export type JobRoles = {
```
Comment (shaunganley) [inline src/models/JobRoles.ts]: not needed

File: src/models/JobRolesResponse.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export type JobRolesResponse = {
```
Comment (shaunganley) [inline src/models/JobRolesResponse.ts:1]: does this not have an ID in it so we can link to the details page later?

File: test/unit/services/JobRoleServiceTest.ts [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        expect(results[0].location).to.deep.equal(jobRolesResponse.location);
+        expect(results[0].roleName).to.deep.equal(jobRolesResponse.roleName);
+        expect(results[0].status).to.deep.equal(jobRolesResponse.status);
+      })
+
+      it('should throw exception when 500 error returned from axios', async () => {
+        mock.onGet("http://localhost:8080/api/job-roles").reply(500);
+
+        try {
+          await getJobRoles();
```
Comment (shaunganley) [inline test/unit/services/JobRoleServiceTest.ts:44]: add a failure call below this line so the test will fail if this call doesn't throw an error

**Review by shaunganley — CHANGES_REQUESTED** [review]

**Review by shaunganley — APPROVED** [review]

**Pull request merged**